### PR TITLE
Implements drop table/database 

### PR DESCRIPTION
**Pull request by dantengsky**
## Summary

Implements store functionalities for DDL stmt "drop table / database"

## Changelog

- Add drop_table/database methods for MemEngine
- Add UT cases for MemEngine accordingly
- Implements related methods  of ActionHandler
- Add some stateless test cases accordingly

## Related Issues

None

## Test Plan

Unit Tests
Stateless Tests

### CODE REVIEW

**Comment by dantengsky** [comment]
pls do not merge, I am adding some extra stateless test cases;

**Comment by dantengsky** [comment]
@BohuTANG @drmingdrmer  
This PR is ready to review, thanks!

**Comment by dantengsky** [comment]
Are there any magic words that I can use in comments to auto-assign reviewers? or did I missed something? I can not request members for review by using the "Reviewers" button of the GitHub PR page.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
@dantengsky 
Hi, your maintainer invitations is pending, please confirm it via your email.

**Comment by dantengsky** [comment]
> @dantengsky
> Hi, your maintainer invitations is pending, please confirm it via your email.

Got it, really appreciate it!!

**Comment by drmingdrmer** [comment]
@dantengsky Sorry that I missed this mention yesterday. I'm gonna have a look ASAP

File: fusestore/store/src/engine/mem_engine.rs
Comment (drmingdrmer) [inline fusestore/store/src/engine/mem_engine.rs:131]: @dantengsky as we discussed the last time when we met, I supposed that in your design the database and tables are stored as a file in DFS, but not a metadata record.

Thus `engine/mem_engine.rs` will be removed finally. It does not reflect the architecture anymore.
The only reason it has been kept is to let the tests pass.

E.g., an `ActionHandler` should get rid of the `MemEngine` ASAP

Comment (dantengsky) [reply fusestore/store/src/engine/mem_engine.rs:131]: Yeah,  metadata will be stored as files in DFS. But before PR 634 is closed, mem_engine is still needed. 

> The only reason it has been kept is to let the tests pass.

This PR is also mainly for the convenience of stateless test.

**Pull request merged**
